### PR TITLE
Update Tf pin to 2023/02/01

### PR DESCRIPTION
**Pull request by JackCaoG**
This is to fix
1. OOM issue on TPU v4-8 (need to wait for tmr's libtpu)
2. nonzero issue on TPU

### CODE REVIEW

**Comment by JackCaoG** [comment]
New error is 
```
tensorflow/core/kernels/fft_ops.cc: In member function ‘std::optional<std::unique_ptr<stream_executor::fft::Plan> > tensorflow::{anonymous}::FftPlanCache::Extract(const Key&)’:
tensorflow/core/kernels/fft_ops.cc:458:12: error: could not convert ‘value’ from ‘tensorflow::{anonymous}::FftPlanCache::Value {aka std::unique_ptr<stream_executor::fft::Plan>}’ to ‘std::optional<std::unique_ptr<stream_executor::fft::Plan> >’
     return value;
```

**Comment by JackCaoG** [comment]
I frankly don't understand why we can't convert `std::unique_ptr<stream_executor::fft::Plan>` to `std::optional<std::unique_ptr<stream_executor::fft::Plan> >`.  Change is from https://github.com/tensorflow/tensorflow/commit/877e6ecbd43957d8e7313efb61e962149f6231cb. Worse case I will have a patch to undo the optional, or rewrite it using a pointer or something...

@ymwangg any insights? Removing

**Comment by ymwangg** [comment]
Maybe try `return std::move(value)`? I saw similar errors before and it seems to be related to compiler versions.

**Comment by JackCaoG** [comment]
Thanks @ymwangg I talked with cl author and we come up with a similar fix, not the new failure is
```
ERROR: /tmp/pytorch/xla/third_party/tensorflow/tensorflow/compiler/xla/runtime/BUILD:268:11: Compiling tensorflow/compiler/xla/runtime/ffi.cc failed: (Exit 1): crosstool_wrapper_driver_is_not_gcc failed: error executing command external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc -MD -MF bazel-out/k8-opt/bin/tensorflow/compiler/xla/runtime/_objs/ffi/ffi.pic.d ... (remaining 167 arguments skipped)
In file included from /usr/include/c++/7/memory:64:0,
                 from ./tensorflow/compiler/xla/runtime/ffi.h:19,
                 from tensorflow/compiler/xla/runtime/ffi.cc:16:
/usr/include/c++/7/bits/stl_construct.h: In instantiation of ‘void std::_Construct(_T1*, _Args&& ...) [with _T1 = xla::runtime::ffi::OwnedFfiState; _Args = {const xla::runtime::ffi::OwnedFfiState&}]’:
/usr/include/c++/7/bits/stl_uninitialized.h:83:18:   required from ‘static _ForwardIterator std::__uninitialized_copy<_TrivialValueTypes>::__uninit_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const xla::runtime::ffi::OwnedFfiState*, std::vector<xla::runtime::ffi::OwnedFfiState> >; _ForwardIterator = xla::runtime::ffi::OwnedFfiState*; bool _TrivialValueTypes = false]’
/usr/include/c++/7/bits/stl_uninitialized.h:134:15:   required from ‘_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const xla::runtime::ffi::OwnedFfiState*, std::vector<xla::runtime::ffi::OwnedFfiState> >; _ForwardIterator = xla::runtime::ffi::OwnedFfiState*]’
/usr/include/c++/7/bits/stl_uninitialized.h:289:37:   required from ‘_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, std::allocator<_Tp>&) [with _InputIterator = __gnu_cxx::__normal_iterator<const xla::runtime::ffi::OwnedFfiState*, std::vector<xla::runtime::ffi::OwnedFfiState> >; _ForwardIterator = xla::runtime::ffi::OwnedFfiState*; _Tp = xla::runtime::ffi::OwnedFfiState]’
/usr/include/c++/7/bits/stl_vector.h:331:31:   required from ‘std::vector<_Tp, _Alloc>::vector(const std::vector<_Tp, _Alloc>&) [with _Tp = xla::runtime::ffi::OwnedFfiState; _Alloc = std::allocator<xla::runtime::ffi::OwnedFfiState>]’
./tensorflow/compiler/xla/runtime/ffi.h:62:8:   required from ‘absl::lts_20220623::internal_statusor::StatusOrData<T>::StatusOrData(std::in_place_t, Args&& ...) [with Args = {xla::runtime::ffi::FfiStateVector&}; T = xla::runtime::ffi::FfiStateVector]’
external/com_google_absl/absl/status/statusor.h:669:55:   required from ‘absl::lts_20220623::StatusOr<T>::StatusOr(std::in_place_t, Args&& ...) [with Args = {xla::runtime::ffi::FfiStateVector&}; T = xla::runtime::ffi::FfiStateVector]’
external/com_google_absl/absl/status/statusor.h:456:52:   required from ‘absl::lts_20220623::StatusOr<T>::StatusOr(U&&) [with U = xla::runtime::ffi::FfiStateVector&; typename std::enable_if<absl::lts_20220623::conjunction<absl::lts_20220623::disjunction<std::is_same<T, typename std::remove_cv<typename std::remove_reference<U&&>::type>::type>, absl::lts_20220623::negation<absl::lts_20220623::disjunction<std::is_same<absl::lts_20220623::StatusOr<T>, typename std::remove_cv<typename std::remove_reference<U&&>::type>::type>, std::is_same<absl::lts_20220623::Status, typename std::remove_cv<typename std::remove_reference<U&&>::type>::type>, std::is_same<std::in_place_t, typename std::remove_cv<typename std::remove_reference<U&&>::type>::type>, absl::lts_20220623::internal_statusor::IsDirectInitializationAmbiguous<T, U&&> > > >, std::is_constructible<_Tp, _Up&&>, std::is_convertible<_Up&&, _Tp>, absl::lts_20220623::disjunction<std::is_same<typename std::remove_cv<typename std::remove_reference<_SrcTuple>::type>::type, T>, absl::lts_20220623::conjunction<absl::lts_20220623::negation<std::is_convertible<U&&, absl::lts_20220623::Status> >, absl::lts_20220623::negation<absl::lts_20220623::internal_statusor::HasConversionOperatorToStatusOr<T, U&&, void> > > > >::value, int>::type <anonymous> = 0; T = xla::runtime::ffi::FfiStateVector]’
tensorflow/compiler/xla/runtime/ffi.cc:383:10:   required from here
/usr/include/c++/7/bits/stl_construct.h:75:7: error: use of deleted function ‘xla::runtime::ffi::OwnedFfiState::OwnedFfiState(const xla::runtime::ffi::OwnedFfiState&)’
     { ::new(static_cast<void*>(__p)) _T1(std::forward<_Args>(__args)...); }
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from tensorflow/compiler/xla/runtime/ffi.cc:16:0:
./tensorflow/compiler/xla/runtime/ffi.h:50:8: note: ‘xla::runtime::ffi::OwnedFfiState::OwnedFfiState(const xla::runtime::ffi::OwnedFfiState&)’ is implicitly deleted because the default definition would be ill-formed:
 struct OwnedFfiState
        ^~~~~~~~~~~~~
./tensorflow/compiler/xla/runtime/ffi.h:50:8: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = XLA_FFI_Module_State; _Dp = xla::runtime::ffi::FfiStateDeleter]’
In file included from /usr/include/c++/7/memory:80:0,
                 from ./tensorflow/compiler/xla/runtime/ffi.h:19,
                 from tensorflow/compiler/xla/runtime/ffi.cc:16:
/usr/include/c++/7/bits/unique_ptr.h:383:7: note: declared here
       unique_ptr(const unique_ptr&) = delete;
       ^~~~~~~~~~
cc1plus: warning: unrecognized command line option ‘-Wno-unknown-warning-option’
```

**Comment by ymwangg** [comment]
This may be something similar. Adding a std::move here should work
`tensorflow/compiler/xla/runtime/ffi.cc:383:10:   required from here`.

**Comment by JackCaoG** [comment]
Thanks @ymwangg after fixing a couple more places, I am able to get gpu to build locally. Let me submit cl internally to fix those so we can drop the patch in next pin update.

File: tf_patches/ffp_gpu.diff [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+diff --git a/tensorflow/compiler/xla/runtime/ffi.cc b/tensorflow/compiler/xla/runtime/ffi.cc
+index 584d2008b11..231abca9c92 100644
+--- a/tensorflow/compiler/xla/runtime/ffi.cc
++++ b/tensorflow/compiler/xla/runtime/ffi.cc
+@@ -380,7 +380,7 @@ absl::StatusOr<FfiStateVector> FfiModulesState::state_vector() const {
+ 
+     return absl::InternalError("Unsupported FFI module state");
+   }
+-  return state_vector;
++  return std::move(state_vector);
```
Comment (steventk-g) [inline tf_patches/ffp_gpu.diff:24]: Nit: Could you add a comment about why `std::move` is needed for us in these cases?

**Review by steventk-g — APPROVED** [review]

**Comment by steventk-g** [comment]
Could you add a comment with simple performance results before/after pin update to verify there's no regression?

**Review by ManfeiBai — APPROVED** [review]

**Comment by ManfeiBai** [comment]
Do we need to check the upstream pytorch tests before merge?

**Comment by JackCaoG** [comment]
> Could you add a comment with simple performance results before/after pin update to verify there's no regression?

It is a compiler bug, I actually already submit the cl internally for review, we should be able to delete this diff in next pin update.

**Review by vanbasten23 — APPROVED** [review]

**Pull request merged**